### PR TITLE
Feat/create whitelist app

### PR DESCRIPTION
**Pull request by Jcalcaldev**
Things to look out for:

- Changed "client" app's name to "sales-client"
- Added whitelist-client app:
  - Uses vanilla React.
- Added whitelist-api app:
  - Uses a postrgres database set up through a docker-compose and TypeORM.
  - Uses NestJS as a framework. 
  - Added a user entity.

### CODE REVIEW

File: apps/whitelist-api/src/app/database/database.module.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  ],
+})
+export class DatabaseModule {
+  static forFeature = TypeOrmModule.forFeature;
+
+  // async onModuleInit(): Promise<void> {
+  //   if (environment.stage === Stages.Testing) {
+  //     await this.clearDatabase();
+  //   }
+  // }
```
Comment (RafaelVidaurre) [inline apps/whitelist-api/src/app/database/database.module.ts:23]: Don't we wanna clear the DB in test environments?

File: apps/whitelist-api/src/environments/environment.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
```
Comment (RafaelVidaurre) [inline apps/whitelist-api/src/environments/environment.ts:1]: How about trying the approach we used in tibiaservers? It used yup schemas which both applied the right typing, were less repetitive and less error prone (if I remember correctly). This is not an isomorphic app so you wouldn't need two files for env vars as we were doing there. But worth having a look

**Review by RafaelVidaurre — APPROVED** [review]
Looking great, check my suggestions and see if its worth applying.

I'd also like you to talk with @gabo2192 about Prisma, this is a good chance to try new tech and I'm not fully satisfied with some things about TypeORM

File: apps/whitelist-api/src/app/database/database.module.ts (hunk 1 above)
Comment (Jcalcaldev) [reply apps/whitelist-api/src/app/database/database.module.ts:23]: Just commented this until we set up a test environment. Will note it down.

File: apps/whitelist-api/src/environments/environment.ts (hunk 2 above)
Comment (Jcalcaldev) [reply apps/whitelist-api/src/environments/environment.ts:1]: Noted!

**Pull request merged**